internal/service/vmservice/vm.go

```diff
@@ -287,6 +327,120 @@ func reconcileVirtualMachineConfig(ctx context.Context, machineScope *scope.Mach
 		}
 	}
 
+	// Additional data disks
```

Comment:
Perhaps, if you can create another function for additional volumes as this raises complexity in the `reconcileVirtualMachineConfig`.